File: 1720.decode-xo-red-array.py
#
# @lc app=leetcode id=1720 lang=python3
#
# [1720] Decode XORed Array
#

# @lc code=start
class Solution:
    def decode(self, encoded: List[int], first: int) -> List[int]:
        encoded[0] ^= first
        for i in range(1, len(encoded)):
            encoded[i] ^= encoded[i-1]
        encoded.insert(0, first)
        return encoded
# @lc code=end

File: cpp/1720.decode-xo-red-array.cpp
/*
 * @lc app=leetcode id=1720 lang=cpp
 *
 * [1720] Decode XORed Array
 */
#include <vector>
using namespace std;
// @lc code=start
class Solution
{
public:
    vector<int> decode(vector<int> &encoded, int first)
    {
        encoded[0] ^= first;
        for (int i = 1, width = encoded.size(); i != width; ++i)
        {
            encoded[i] ^= encoded[i - 1];
        }
        encoded.insert(encoded.begin(), first);
        return encoded;
    }
};
// @lc code=end
